modules/mod_facl.c

```diff
@@ -1092,27 +1102,36 @@ static int facl_fsio_access(pr_fs_t *fs, const char *path, int mode,
 
       if (sys_access(fs, path, mode, uid, gid, suppl_gids) < 0) {
         xerrno = errno;
+
         pr_trace_msg(trace_channel, 6, "normal access check for '%s' "
           "failed: %s", path, strerror(xerrno));
         errno = xerrno;
         return -1;
       }   
 
       return 0;
-
-    } else {
-      return -1;
     }
+
+    errno = xerrno;
+    return -1;
   }
 
   pr_trace_msg(trace_channel, 10,
     "acl(2) returned %d ACL entries for path '%s'", nents, path);
 
-  acls = pcalloc(fs->fs_pool, nents * sizeof(aclent_t));
```

Comment:
These allocations out of the `fs->fs_pool` are a *huge* resource leak; that pool is for the lifetime of the process, and is not cleared frequently.  Thus we instead create a temporary sub-pool, for this task, that we can release/destroy once we're done.